src/datachain/fs/utils.py

```diff
@@ -0,0 +1,27 @@
+from typing import TYPE_CHECKING
+
+from fsspec.implementations.local import LocalFileSystem
+
+if TYPE_CHECKING:
+    from fsspec import AbstractFileSystem
+
+
+def isdir(fs: "AbstractFileSystem", path: str) -> bool:
+    """
+    Returns true if uri points to a directory.
+
+    Extends `fs.isdir` to support special directories on object storages, e.g.:
+    Google creates a zero byte file with the same name as the directory with a trailing
+    slash at the end.
+    """
+    if isinstance(fs, LocalFileSystem):
+        return fs.isdir(path)
+
+    try:
+        info = fs.info(path)
+    except FileNotFoundError:
+        return False
+    else:
+        return info["type"] == "directory" or (
+            info["size"] == 0 and info["type"] == "file" and info["name"].endswith("/")
+        )
```

Comment:
Improving the check for zero-byte file with a trailing slash.

Reply:
🤔 Looks reasonable, but same time looks a bit suspicious 👀
I know this was done by a reason and was checked many times, so I am ok with this 🙃